cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -68,25 +75,15 @@ public void setRepository(FullHttpRequest request, HttpResponder responder,
                             @PathParam("namespace-id") String namespaceId) throws Exception {
     checkSourceControlFeatureFlag();
     NamespaceId namespace = validateNamespaceId(namespaceId);
-    RepositoryConfigRequest repoRequest = getRepositoryConfigRequest(request);
+    RepositoryConfigRequest repoRequest = validateAndGetRepoConfig(request);
 
-    try {
-      if (repoRequest == null || repoRequest.getConfig() == null) {
-        throw new RepositoryConfigValidationException("Repository configuration must be specified.");
-      }
-
-      repoRequest.getConfig().validate();
-      if (repoRequest.shouldTest()) {
-        // TODO: CDAP-20252, add the validate logic once the SourceControlManager module is ready
-      }
-
-      sourceControlService.setRepository(namespace, repoRequest.getConfig());
-      responder.sendJson(HttpResponseStatus.OK,
-                         GSON.toJson(
-                           new SetRepositoryResponse(
-                             String.format("Updated repository configuration for namespace '%s'.", namespaceId))));
-    } catch (RepositoryConfigValidationException e) {
-      responder.sendJson(HttpResponseStatus.BAD_REQUEST, GSON.toJson(new SetRepositoryResponse(e)));
+    if (repoRequest.shouldTest()) {
+      RepositoryManager.validateConfig(secureStore,
+                                       new SourceControlConfig(namespace, repoRequest.getConfig(), cConf));
+      responder.sendStatus(HttpResponseStatus.OK);
```

Comment:
right now it's functionally the same, but it's better if we return here and remove the else clause.

```
if (shouldTest()) {
  ...
  responder.sendStatus();
  return;
}

normal logic here
```

it removes one level of nesting and makes it less likely that we'll accidentally use the responder twice on future edits and refactorings.

Reply:
nice catch, updated to avoid the else clause